### ISSUE TITLE
imap Gmail inbox label remove issue after Ticket rejected via email fetch

### ISSUE THREAD

**Issue by ysoylemez** [osTicket/osTicket#6407]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ ] Do you have any errors in the PHP error log, or javascript console?
* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description
Hello,

We upgraded the system to 1.17.2 version last week. After this version update we are facing some email fetch issue. 
We are using gmail imap to get emails and create ticket. Some emails inbox label is removed by fetch but ticket is not created. I am getting this error when some email is rejected due to there is no user for this sender email address. 

I am tinking the issue may be releated with below;
Example ; I have 5 unread email in inbox. Emails per fetch parameter is 6. First email sender is not registered in the system and it's rejected by fetch. But the system is remove all 5 email labels in the gmail. Remaining 4 emails are not proceed into tickey system. I coudln't find any error logs in the PHP.

I am performing manual process to move all unread emails to inbox again occurs every day to get them into system. 

Do you have any suggestion about how to solve this issue?

![image](https://user-images.githubusercontent.com/100907272/207060127-0ebd6d8c-d671-4094-9c80-c3537f049fc9.png)

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Versions
1.17.2
Admin panel -> Dashboard -> Information which also additionally gives you information about your server.

Also, please include the OS and what version of the OS you're running. As well as your browser and browser version.

**Comment by JediKev** [comment]
@ysoylemez 

Try this:
- https://github.com/osTicket/osTicket/pull/6381

Cheers.

**Comment by ysoylemez** [comment]
Hi JediKev,

Thank you for quick response. I uplaoded this changes the system. The email fetch issue seems fixed.  

I see one other issue on the scp/logs.php page after this changes. When i click one event log, i see below apache error on the server. I couldn't find how can i fix. 

PHP Fatal error:  Declaration of ContentAjaxAPI::log($id) must be compatible with ApiController::log($title, $msg, $level = LOG_WARN) in /var/www/support/include/ajax.content.php on line 22, referer: https://support.snitechnology.net/scp/logs.php?&p=1

**Comment by ysoylemez** [comment]
Hi, @JediKev 

Thank you for quick response. I uplaoded this changes the system. I will monitor a few days. I will update the status.  

I see one other issue on the scp/logs.php page after this changes. When i click one event log, i see below apache error on the server. I couldn't find how can i fix. 

PHP Fatal error:  Declaration of ContentAjaxAPI::log($id) must be compatible with ApiController::log($title, $msg, $level = LOG_WARN) in /var/www/support/include/ajax.content.php on line 22, referer: https://support.company.com/scp/logs.php?&p=1

**Comment by ysoylemez** [comment]
Hi @JediKev ,

It seems the issue not solved yet. I see some emails not pulled by system. I manually move the emails to inbox folder and pulled by system.  When i checked the osticket system logs, i saw below warning. 
The logs and email receivign times are same. This error may be cause. 

Error: Mail Fetcher : the single id was not found in response

Thanks,

**Comment by JediKev** [comment]
@ysoylemez 

To fix the fatal error you need this:
- https://github.com/osTicket/osTicket/pull/6399

As for the `Mail Fetcher` error I have not seen that one before. Maybe that was a one-off thing? Try again and see if you run into the same issues.

Cheers.

**Comment by ysoylemez** [comment]
Hi @JediKev ,

I uploaded this changes. I will monitor the email and system. I will update the status.

Mail Fetcher is the subject of the log messages. 
![image](https://user-images.githubusercontent.com/100907272/207376229-c6ad8d09-3205-495b-aeac-1b89cb108927.png)


Thanks,

**Comment by ysoylemez** [comment]
Hi @JediKev 

I can see this error a few times yesterday and today. The mail is not processed and inbox label is removed on the gmail. I manually moved emails to inbox again. 

![image](https://user-images.githubusercontent.com/100907272/207853383-6b1ca1fc-443e-4855-8710-42f13d031ad7.png)

**Comment by tauthement** [comment]
I am getting the same error "single id" error message. Several emails a day are being archived and left unread without a label. They never get retrieved to the Helpdesk. They just sit in "All Mail" with no label. I have to manually move them to the Inbox, but sometimes the same issue occurs and they don't get pulled. I have to move them to the Inbox several times before they actually get pulled into the helpdesk.

**Comment by tauthement** [comment]
To add, searching the code for the error message finds this message to be within [Imap.php](https://github.com/osTicket/osTicket/blob/8fbc7eea6a473ee76e68d7d1a0a36a58c56b5939/include/laminas-mail/src/Protocol/Imap.php#L616) in laminas-mail.

**Comment by LH-Eelko** [comment]
I can confirm this exact behavior with our own setup.

- We're using Gmail for e-mail handling
- Sometimes emails are being moved from "_Inbox_" to "_All mail_" without getting labeled or getting marked as read
- We also see the "the single id was not found in response" message in our logging
- There are no noticeable errors in our (server) error logs
- When setting log level to "Debug" (or any other level fwiw), no other info is being logged that could lead to a possible problem

osTicket 1.17.2 (upgraded from 1.16.3)
PHP 8.1
nginx/1.23.3
MariaDB 10.6.10
5 minute cron/fetch interval

Since 1.17.x also seems to break the plug-ins we're using, we had them disabled all (so no other things that could interfere with our [email] workflow).

[edit] Probably also interesting and good to know (and what also has been mentioned, I think); when you move the message back to "_Inbox_" it usually does get processed in the next run (sometimes you have to try multiple times).

**Comment by tauthement** [comment]
@LH-Eelko For the plugin issue, try downgrading PHP to 8.0. I had the same issue until I found out that 8.1 is not fully supported yet.

**Comment by LH-Eelko** [comment]
> @LH-Eelko For the plugin issue, try downgrading PHP to 8.0. I had the same issue until I found out that 8.1 is not fully supported yet.

I rewrote some parts of the plugins we are using to be compatible with PHP 8.1 (and osTicket 1.17.x). The new instances part of osTicket really seems to affect older plugins.

We have one rather imporant plugin left to debug, but I'm sure we'll get that to work too.

Will try your suggestion though! Could save us some work which would be nice (although I'm pretty sure it's the instances part that is preventing the plugin from working correctly). :)

**Comment by tauthement** [comment]
@LH-Eelko For what it's worth, I'm using some old plugins and they all work with PHP 8.0 and osTicket 1.17. I didn't have to change anything code-wise.

**Comment by JediKev** [comment]
@tauthement PHP 8.1 is supported; I don’t know where you got the information it isn’t. PHP 8.2 is not yet supported as it’s new but 8.0 and 8.1 are definitely supported with 1.17.x.

@LH-Eelko You are supposed to update any installed plugins to the latest builds when upgrading osTicket. As core code changes so do plugins.

I would suggest to just wait until v1.17.3 and the next build of the plugin is released and then retest. For what it’s worth I use v1.17.2, PHP 8.1, and the latest build of the OAuth2 plugin with both Gmail and O365 and cannot replicate this issue.

**Comment by LH-Eelko** [comment]
@tauthement PHP 8.0 didn't solve our issues with the specific plugin. Was worth the try though (but as said, I didn't have any hope since I expect that the plugin code probably isn't compatible with 1.17.x). :)

@JediKev thanks for the confirmation! :) I fully understand what you're saying. The plugin is rather old and not updated. It doesn't surprise me that it no longer works with an upgraded core. Most likely an updated version of osTicket is not going to fix the problem either. I'll look into fixing the plugin as soon as I got some more time on my hands. Pretty sure it needs some PHP 8.1 love anyway.

As for this part of your reaction;

`For what it’s worth I use v1.17.2, PHP 8.1, and the latest build of the OAuth2 plugin with both Gmail and O365 and cannot replicate this issue.`

Is that regarding the little plugin discussion or the actual problem (the single id message / osTicket not importing certain emails)? Might have hijacked the thread a bit with my plugin comment which was not my intention (and sorry if I did).

**Comment by tauthement** [comment]
@JediKev The official [README](https://github.com/osTicket/osTicket/blob/develop/README.md) does not list 8.1, only 8.0. Only going by the official documentation and personal experience. When I tried 8.1, I encountered issues with third-party plugins that haven't been updated and unlikely won't be so I'm just stuck on 8.0 for now.

Regardless, IMAP fetching is the issue at hand. PHP and plugin support can be hashed out elsewhere.

**Comment by JediKev** [comment]
@tauthement 

Yes, we still need to update our documentation and README but PHP 8.0 and 8.1 are both fully supported with the latest core version and latest builds of plugins.

I am still unable to replicate the fetching issue so I’m unsure what’s going on in your case. Are you able to create a completely new email account and test that in osTicket? It could be an issue with legacy accounts.

Cheers.

**Comment by tauthement** [comment]
Hi @JediKev,

Sorry for much delayed response. I forgot about this after returning from the Christmas holidays. We still have the issue and still have to monitor the email account to push emails through. 

When you mentioned legacy accounts, do you mean using IMAP? We currently use IMAP with Gmail. Does OAuth2 authorization replace IMAP access to a mailbox?

**Comment by jasonhoward7** [comment]
@JediKev  - I am having this same IMAP issue as @LH-Eelko.

My setup:
osTicket 1.17.2 (upgraded from 1.14.1)
PHP 8.1.2
apache/2.4.52
mysql 8.0.32
1 minute cron/fetch interval

• We're using Gmail for e-mail handling
• Sometimes emails are being moved from "Inbox" to "All mail" without getting labeled or getting marked as read
• We also see the "the single id was not found in response" message in our logging

I'd love to hear any suggestions on testing. I'm happy to get into the core and test if that would help - though  I'd appreciate any suggestions on where to look.

**Comment by milanteo** [comment]
> Sometimes emails are being moved from "Inbox" to "All mail" without getting labeled or getting marked as read

I can confirm this strange behavior, this is our server configuration: 

![immagine](https://user-images.githubusercontent.com/24292225/222369013-d03276fb-1868-4c5d-b27a-2ee55b8a19aa.png)

Setting up email deletion after processing has apparently solved it, but no error logs like this in our logging page:

> We also see the "the single id was not found in response" message in our logging

I'm available to provide further infos if needed,

Thank you!

**UPDATE:** Email are no longer being deleted, tickets are being processed correctly and emails stay in 'All Messages' tab

**Comment by JediKev** [comment]
@milanteo 

Have you applied the above changes I referenced? There are multiple pull requests you can apply and retest.

Cheers.

**Comment by triagehhardy** [comment]
Hi @JediKev 

This issue does not seem to be resolved, I update to 1.17.3 a week ago and the issue remains. I am still missing emails that are not fetched.

I still receive the Mail Fetcher errors:

![image](https://user-images.githubusercontent.com/14201137/228584032-34aa252d-4af7-4940-8644-4d5b88d82ded.png)

Here is my system information:

![image](https://user-images.githubusercontent.com/14201137/228584374-b261dd5e-80cb-48a3-9378-80120c21f077.png)

**Comment by tauthement** [comment]
@triagehhardy I'm still having the same problem as well. I've had to to move an email back to the inbox three times so far this morning and it still hasn't been fetched. I have to babysit an email account all day; a huge waste of time. This didn't happen before the switch to laminas-mail.

**Comment by JediKev** [comment]
@tauthement @triagehhardy 

Unfortunately, I'm unable to replicate this issue and I've seen virtually no more reports of this since v1.17.3 (v1.17.2 I think actually) and the latest build of the OAuth2 plugin (except your two reports). I would highly recommend re-uploading the v1.17.3 code, making sure you have the latest build of the OAuth2 plugin installed, and restarting the webserver + PHP-FPM (if you're running it; this will clear any file cache, etc.).

If you still run into issues, you may just need to wait until we do the next laminas-mail update. They have a couple new versions since the version we are using so this should bring more improvements. This isn't scheduled for a bit though so it will take some time before this gets released. So please stay tuned!

Cheers.

**Comment by tauthement** [comment]
@JediKev Thanks for the response. I'm running the 0.6 version of the OAuth2 plugin, which seems to be the latest in the osTicket-plugins repo. However, this was happening before I started using the OAuth2 plugin for the email account. I was using Basic Authentication before. 

Oddly enough, I have two other osTicket instances installed on the same server for other departments and neither have the same issue. However, those are very, very low traffic. They only see a few tickets a week, if lucky. 

On the instance having the problem, I also have a second email address connected to it that doesn't have any unread emails in it that were supposed to be fetched. That account is very low volume. The account having the issue has a higher volume and can get 50-100 emails per day. I think what I'll try doing is removing the account and re-adding the account during our maintenance hours tonight and see if that helps. It's a little strange to me that it's only happening to one account across several instances.

Looking forward to the new version.

**Comment by triagehhardy** [comment]
@JediKev @tauthement I also get the sense this is a volume issue, my account can get 100-150 emails per day. When I miss emails it tends to be during peak times.

**Comment by JediKev** [comment]
@tauthement @triagehhardy 

Out of curiosity, what are the values for your Fetch Frequency and Mails Per Fetch settings? Also, what is your cron job/scheduled task set to? Are you Archiving or Deleting mail after fetch? What email provider are you using?

Cheers.

**Comment by triagehhardy** [comment]
@JediKev @tauthement 

Fetch Frequency is 3 minutes, Mails per fetch is 30, Cron is set to 3 minutes.

I am archiving to another label after the fetch, the emails that are missed are generally removed from inbox but are not archived and marked as read.

Gmail for email provider.

Just fyi i did not get this issue until 1.17.

**Comment by tauthement** [comment]
Fetch frequency: 5 minutes
Emails per fetch: 10
Cron jobs: every 5 minutes

I'm using Gmail and I'm archiving to a label called "Helpdesk Fetched". Similar to @triagehhardy, the missed emails are removed from the inbox, unlabeled, and marked as read.

**Comment by JediKev** [comment]
@tauthement @triagehhardy 

Definitely weird. I use Gmail, use pretty much the exact same setup, and don't run into such issues. Like I said if you perform the steps I listed above and still run into issues you may just need to wait until we update laminas-mail in our codebase.

Cheers.

**Comment by tauthement** [comment]
@JediKev Do you think the user specified to run the cron job matters? I currently have the cron jobs set to run under the apache user.

**Comment by JediKev** [comment]
@tauthement 

The apache user should be fine; as long as the user has privileges to execute the PHP script successfully (which apache user should have). I usually edit /etc/crontab and just keep the user blank (empty).

Cheers.

**Comment by tauthement** [comment]
Thanks. I figured that would be the case, but wanted to make sure. I currently use the following and use "crontab -u apache -e" to edit it so it's under the apache user without having to specify the user.

*/5 * * * * /usr/bin/php /var/www/helpdesk.nicholls.edu/api/cron.php

I'll try removing the account tonight like I mentioned and test it out for a few days. Not hopeful, but it's worth a shot.

**Comment by milanteo** [comment]
I can confirm that also after updating to 1.17.3 I'm encountering fetching and missing labels issues in Archiving mode and correct fetching but missing deletion with Deleting mode.

To my previous provided details I can add that we're running OsTicket on a docker instance based on the official PHP 8.1 apache image, is this the case for some others experiencing the same issue?

Thanks!

**Comment by triagehhardy** [comment]
@milanteo I am not running a docker instance, so that doesn't seem to be the issue.

**Comment by JediKev** [comment]
@milanteo 

Also, there is no official docker instance; we do not use docker at all but plan to with v2.0. Someone stole our namespace and created their own instance; we are in the process of recovering our namespace. We do not recommend the docker container as many people have reported various issues with it.

Cheers.

**Comment by tauthement** [comment]
I knew it was a stretch, but I deleted the email account within osTicket and recreated it last night. This morning I had two unfetched emails in the All Mail folder with no label.

**Comment by JediKev** [comment]
@tauthement 

Then it seems like you'll definitely have to wait for the laminas-mail update. Wish I could replicate this..

Cheers.

**Comment by tauthement** [comment]
Not sure if this will be insightful, but here's a Stack Overflow thread about the same error message. It references zend-mail, which was renamed to laminas-mail.

https://stackoverflow.com/questions/4183080/zend-mail-movemessage-exception
https://framework.zend.com/issues/browse/ZF-5655

**Comment by milanteo** [comment]
> @milanteo
> 
> Also, there is no official docker instance; we do not use docker at all but plan to with v2.0. Someone stole our namespace and created their own instance; we are in the process of recovering our namespace. We do not recommend the docker container as many people have reported various issues with it.
> 
> Cheers.

We built our docker image ourselves based on the official PHP one after seeing that the image you mentioned wasn't receiving updates for years, glad we didn't choose that image now that I know it wasn't even an official image from you, thanks for the info.

At the moment we're experiencing no other issues / strange system logs, but I can confirm that our email traffic is very similar to the ones mentioned before, starting from around 50 daily mails to 100's peak during busy days.

Regards

**Comment by tauthement** [comment]
On Friday morning, I set the "Fetch Emails" setting to "Do Nothing (not recommended)" and I haven't had a skipped email since. I'm still keeping an eye on it, but I assume the issue is with the archive feature.

**Comment by triagehhardy** [comment]
@tauthement 

Any skipped mail yet? I may turn my archive off if that's the case.

**Comment by tauthement** [comment]
@triagehhardy No skipped emails so far.

**Comment by tauthement** [comment]
Just a quick check in. Since I've switched the setting to "Do Nothing", every email has been picked up by the fetcher. It seems like the issue is with the Archive feature.

**Comment by triagehhardy** [comment]
Yup, same for me.

**Comment by randygitto** [comment]
Hi, 

I have been having this issue for a long time (over a year) with some very upset customers thinking we are ignoring them. I finally found an error code that led me here after a google search "the single id was not found in response" I was able to witness this error and the email not being picked up and then archived as NEW. I re-inboxed the email and it went through okay.  I have been doing that for about a year now. 

 I just disabled the archiving for the inbox, and I am hoping it helps.  Has there been any fix? I am on 1.18.1

**Comment by tauthement** [comment]
I think I finally fixed this issue on my end, but I was waiting to add my comment until I gave it time to confirm that the issue didn't rear its head again. It's been about a month and while I still see the "single id was not found in response" message in the logs from time to time, I haven't had any emails skipped with archiving enabled.

I applied [this fix](https://forum.osticket.com/d/103604-imap-with-oauth2-on-mso-cron-error-setconfigclass-on-null/2) from JediKev that he posted on the osTicket forums. Hope this resolves the issue on your end as well.

**Comment by randygitto** [comment]
I am using legacy authentication with gmail, would that fix apply to me?
Alternatively, how has the cron been when you didn't archive the emails?

**Comment by tauthement** [comment]
The fix relates to the plugin instance so the way you authenticate shouldn't matter. From what I remember when I applied the fix, some of the settings were pointing to the wrong instance. I tried removing and adding instances countless times to see if it would help the issue so I probably introduced some junk entries in there that never got cleared. Maybe @JediKev can give more insight on that fix.

The cron job worked fine when I didn't archive the emails. It just kept all of the emails in the Inbox and marked them as read. The helpdesk was pulling all emails instead of skipping some. That was my solution for a while until I applied the mentioned fix.

**Comment by randygitto** [comment]
> The fix relates to the plugin instance so the way you authenticate shouldn't matter. From what I remember when I applied the fix, some of the settings were pointing to the wrong instance. I tried removing and adding instances countless times to see if it would help the issue so I probably introduced some junk entries in there that never got cleared. Maybe @JediKev can give more insight on that fix.
> 
> The cron job worked fine when I didn't archive the emails. It just kept all of the emails in the Inbox and marked them as read. The helpdesk was pulling all emails instead of skipping some. That was my solution for a while until I applied the mentioned fix.

When he said "Then make sure you delete the associated Plugin Instance record either through the _plugin_instance table in db or through the UI. "   I assume the UI is the oath2 plugin, which I am not using. Do you know if that is correct?

**Comment by tauthement** [comment]
Yes, you are correct. So I guess this wouldn't apply to you since you're using legacy authentication. However, you may want to look into using OAuth 2 with Gmail since Google will [only allow access to apps using OAuth](https://workspaceupdates.googleblog.com/2023/09/winding-down-google-sync-and-less-secure-apps-support.html) on September 30, 2024. 

Something else I changed was the name of my label from "Helpdesk Fetched" to just "Helpdesk", eliminating any spaces. Take this advice with a grain of salt because I'm not entirely familiar with the osTicket codebase, but my thought process is that maybe there's some kind of string parsing bug somewhere. I changed the label name at the same time I re-added the plugin instance so it's impossible for me to know if that helped anything.

**Comment by tauthement** [comment]
It looks like I spoke too soon. I got in to work today and someone asked us for an update on a ticket they sent in through email. Lo and behold, the email was still unread, but not in the Inbox. So I'm just switching the archive setting back to "Do Nothing." I don't know what else to do at this point other than let the read emails sit in the Inbox. I'm not going to babysit an email account all day. I have actual work to do.

**Comment by JediKev** [comment]
@tauthement 

1. Do these issues happen with specific emails from specific senders or seemingly random with really any email/sender? If the former, what type of email and what email provider are they using to generate the email?
2. I don't think you ever mentioned this and I apologize if you have, but do you use Gmail or O365 or something else?
3. Do you have a different email address you can setup and test with to see if maybe it's an issue specifically with that one email account?
4. Can you try upgrading your `/path/to/osTicket/include/laminas-mail` package with [the latest from their repo](https://github.com/laminas/laminas-mail/releases/tag/2.25.1)? Unfortunately, I cannot replicate this so I can't "test" any potential fixes. Also, I do not know if they have any breaking changes with this version since the version we are currently using but please test this in a non-production environment in case there are breaking changes that cause further issues.

Cheers.

**Comment by tauthement** [comment]
1. Seemingly random. All emails are from our organization's domain. 
2. We use Gmail Workspace.
3. I have two email accounts setup, but the other is very low traffic. It was used frequently at one point, but rarely anymore. I haven't seen any unread emails on there. I have a few other helpdesk instances, but those are low volume as well and I haven't noticed the issue on those so far.  
4. I can give it a shot in our test instance, but I think it'll be hard to reproduce in there. I'll have to write a script to send a bunch of emails to it at once. Our production environment can get around 100-200 emails per day between new emails and replies. I'm wondering if the volume of emails we get lends itself to some of the emails getting skipped. I have the "Emails Per Fetch" setting set to 30, but we definitely don't get that many at once. I can try upping the value.

I did some digging in the osTicket System Logs page and confirmed that 3 emails which came in around the same time (so likely on the same cron pull) and got skipped had the "the single id was not found in response" error message in the System Logs. Laminas-mail [throws this exception](https://github.com/osTicket/osTicket/blob/4689926b2d3d25754f0ddcf8d4e181a2817f6d56/include/laminas-mail/src/Protocol/Imap.php#L635-L637) if $to === null and $from is not an array. 

Using the "Show Original" feature in Gmail, I can see that the To field is not empty or null, and the From field only contains one address. If you would like, I can send you a copy of the EML file privately, but I'd rather not share it here publicly.

**Comment by JediKev** [comment]
@tauthement 

Yes, I'm aware where the error comes from and why laminas-mail throws the error but I simply cannot replicate the issue via normal fetch operations in any of my instances. Even though the To and From headers look correct in the actual email it appears as though the email content can't be fetched properly so it's unable to grab those headers and therefore those checks fail returning the error. I will say that I use multiple instances of Gmail (normal and Workspace) and do not experience such issues with my setups; although they are all low traffic. We also do not have reports of this in our cloud solution where we have customers using Google Workspace with high volume of mail and are moving to folders post-fetch. So it's very hard to pinpoint the root of the issue.

I don't think me receiving a copy of the email will help as there is no way for me to fetch the original raw email like normal; I'd have to pipe it which will not replicate the issue as I believe this is solely occurring with fetching via IMAP/POP3.

I can't tell if it's an issue with the actual fetch operation where the connection gets borked/corrupt causing issues with fetching mail content or if the archive method of moving to folder messes with the array of unique ids we grab from the configured mail folder before individual mail fetching/parsing begins or what. It's very hard to pinpoint and track down so there is no telling right now without further debug info from someone who can actually replicate the issue.

A good debug test would be to add `error_log($this->getRawHeader($i) . $this->getRawContent($i));` [above the `return` statement here](https://github.com/osTicket/osTicket/blob/develop/include/class.mail.php#L518), let the system attempt to fetch a problematic email, and look at your PHP error logs to see if a specific error message was logged. We've seen with some O365 accounts that this will return an actual error message directly from the mailserver saying there is an issue with the email and it cannot be fetched. So this route may shed more light in your case. ****IMPORTANT NOTE:** This will log the _full raw content_ for _every_ email it fetches so do this during non-production/low traffic hours as it will fill up your logs if you aren't careful.

Cheers.

**Comment by tauthement** [comment]
Seeing as this only happens when the Fetched Emails setting is set to "Archive - Move to Folder", I'm leaning toward it being an issue with the archiving method. When I set the Fetched Emails setting to "Do Nothing" and move those skipped emails back to the Inbox, it fetches the emails without a problem. Setting the feature to "Do Nothing" solves the problem for now. 

I'm aware this is a hard problem to track down. I'm just trying to give you all of the information I have.

**Comment by randygitto** [comment]
Hi Kev, Tauthement,

I had this issue since 2022, on 3 different versions, and spent a TON of hours trying to debug. I am on shared hosting. I disabled all plugins and played with all Gmail and OST settings combiniation, and nothing helped. I even did a clean install of OST. The only thing I know is that this issues came with the the new email system. This bug is so random. Right now, not archiving fixed the problem and I don't have upset customers.  

I get notification outside of office hours and I could test this after hours. Is there a way I can run the laminas-mail outside of OSticket?

**Comment by tauthement** [comment]
Same here. It all started with the switchover to laminas-mail. Everything worked fine before then.

**Comment by JediKev** [comment]
@randygitto 

Of course, setup the package on the server and write your own scripts to use laminas-mail. You can reference their docs for how to use the package.

Cheers.

**Comment by LH-Eelko** [comment]
Just here to confirm that we also recently switched to "Do nothing (not recommended)" instead of  "Archive - move to folder" for the remote mailbox settings. Haven't missed an email since making the switch, where that used to be at least one missed email per day.

Other settings have not changed (still using legacy basic authentication, for example). 

Like @tauthement said, we're just trying to provide as much information as we can. :)

**Comment by TonyTheDarkLord** [comment]
I'm having the same issue, the last year I updated from 1.10 to 1.18 and this happens every day, in 1.10 never happened this(I updated only because every time I ask something in the community I got ignored cause I wasn't in the latest version). It's very strange, I use Gmail to receive the emails, I have a cron task to run every 2 minutes and max fetch 20, also to move to a label called "ticketsabiertos".

The new behavior moves the email to the label but it lefts unread and got a variety of errors like: "the single id was not found in response" and "Unable to create new ticket :user Incomplete client information".
Cause of this, every day **I had to open the Gmail account, and in the search box I use "is: unread" and every mail that is in the label unread I move it to the inbox, and in the next fetch the mail enters the system, BUT not always works the first time, sometimes the same email I have to move it to the inbox 2 or 3 times until enters**. So, I don't understand Why the very first fetch not catch the email correctly, Its very annoying and in the last 2 months more and more mails got this errors, 80% of the emails with errors are from users that are active already in the platform and some are responses.

In the last month we got more than 1000 tickets in the platform, and about 60 emails got this problem, it seems that's not a lot but I implemented this in a University cause I got lots of recommendations at that time, but now I'm waiting for a fix to this, and my manager are not so happy with this, cause if I don't remember to check the email account lots of users complaint. I know that it's free and I really appreciate it as any useful tool and more being open source, but I won't be all my life checking this gmail account, so my manager already request me to test alternatives if this won't get fixed in the next months.

**Comment by tauthement** [comment]
@TonyTheDarkLord Set the Fetched Emails setting to "Do nothing (not recommended)" for now. Emails will accumulate in the Inbox, but they'll at least all be read by osTicket. You can always go into the Inbox later and archive the emails if you want to clean up the inbox.

That's the workaround until this can actually get fixed.

**Comment by TonyTheDarkLord** [comment]
> @TonyTheDarkLord Set the Fetched Emails setting to "Do nothing (not recommended)" for now. Emails will accumulate in the Inbox, but they'll at least all be read by osTicket. You can always go into the Inbox later and archive the emails if you want to clean up the inbox.
> 
> That's the workaround until this can actually get fixed.

Thanks for the information, I will report back if happens again with this setting.
Thanks a lot

**Comment by TonyTheDarkLord** [comment]
> @TonyTheDarkLord Set the Fetched Emails setting to "Do nothing (not recommended)" for now. Emails will accumulate in the Inbox, but they'll at least all be read by osTicket. You can always go into the Inbox later and archive the emails if you want to clean up the inbox.
> 
> That's the workaround until this can actually get fixed.

I can confirm, "do nothing" do the trick, thanks a lot!

**Comment by chufrog** [comment]
We encounter the same issue mentioned.
Gmail Word space, v1.18.1, happen in random email.
We changed the "Fetched Emails" to "do nothing" now.
Hope this issue can be resolved soon.

**Comment by ccl100** [comment]
I run into the same issue with GMAIL. Changing to "Do Nothing" now, hoping this will do it.

**Comment by nicoletta-maia** [comment]
Same problem here too, we tried the “Do nothing” workaround but now the banned emails and those rejected by one of our filters with Reject Ticket action keep getting fetched at every round of the cron; any suggestion?

**Comment by tauthement** [comment]
Not sure what you're filtering on, but you may need to setup filters in the email account, if possible. We have email filters to block any emails from outside our domain.

**Comment by nicoletta-maia** [comment]
Thanks @tauthement, I'll open a new issue with more details because I think I'm offtopic here, even though my issue is a consequence of the workaround suggested in this thread
